### PR TITLE
Collections 699 - PairingIterator

### PR DESCRIPTION
**Pull request by Fival85**
Hello, i implemented the improvement [COLLECTIONS-699](https://issues.apache.org/jira/projects/COLLECTIONS/issues/COLLECTIONS-699). Please review it. Attention: to implement the improvement i must set the dependency common-lang3 to provided. Is this a good idea? We could use a extra dto for that, but i think thats not the way the common libs should be implemented?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -457,7 +457,7 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-lang3</artifactId>
       <version>3.9</version>
-      <scope>test</scope>
+      <scope>provided</scope>
```
Comment (michael-o) [inline pom.xml]: This scope is just wrong!

**Comment by michael-o** [comment]
@garydgregory Why should tuples be part of Collections? The concept is so generic that it is not really tied to collections...

**Comment by Fival85** [comment]
Thank you for your reviews!

I understand tha part with the dependy. What is your opinion to change it from Pair<L,R> to the array resolution. Then the iterator will return an array of items of type E, where each item is one of the child iterators.

**Comment by garydgregory** [comment]
An array of E is not right since that assumes both integrators are of the same parameter type.

**Comment by garydgregory** [comment]
Maybe you just need a light weight holder class that is like a Pair...

**Comment by garydgregory** [comment]
Hm... the light-weight way about this might be an inner class called Entry (or Pair) with two public final instance variables and a ctor. Use that instead of a Lang Pair.

**Comment by Fival85** [comment]
Sure, make a own "pair"-class will be a possibility, but it has the disadvantage that the number of iterator-types are limited (two if we you use a pair eg). 
The solution with the array (or List) is that we could retrieve as many iteratos the user want, but all must have the same type. 

What is the better improvement:

1. iterate over all as many child iterators of the same type
2. iterate over two child iterators with different types

I think mostly the developer would do something with different types. But the limitation of the two types isnt nice.

So, do we need this feature with the light-weight way of @garydgregory ? Otherwise the ticket COLLECTIONS-699 should be closed.

Thank you

**Comment by Fival85** [comment]
i will change the solution and youse a dto to return the iterators result. i think map.entry is a good choice, because its part of the jdk and is useable to the map api

**Pull request closed without merging**